### ISSUE TITLE
The file has been modified and saved however the buffer is not updated automatically

### ISSUE THREAD

**Issue by faruthem** [zed-industries/zed#14991]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

I have a file, specifically I am working with python, when I save the file on an external drive ZED does not update its status to saved even though the file has been saved. Something similar happens if I open a new file and I want to save it specifically on some external media (disk), the status of my file is not updated

### Environment

OS: Fedora 40 KDE Plasma
Memory: 8GB
CPU: Intel i7-4710HQ (8) @ 3.500GHz 
ZED: 0.144.4

### If applicable, add mockups / screenshots to help explain present your vision of the feature

![Captura de pantalla_20240722_180754](https://github.com/user-attachments/assets/3db6adfb-3b79-4f66-84cd-057e9b4f0739)

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary><pre>

No problems in workspace
</pre></details>

**Comment by ConradIrwin** [comment]
@faruthem Do you see errors about `too many open files` in the Zed log (`~/.local/share/zed/logs`). It sounds like we're not able to watch all your files.

**Comment by faruthem** [comment]
Hello, the specific error is the following: when trying to save the changes to my file, only the yellow circle appears, my file is saved correctly but ZED does not update the status of my file to saved.

**Comment by ConradIrwin** [comment]
Right. The way this is supposed to work is that Zed writes the file during Save. Then `inotify` tells Zed that the file has changed and so we update the indicator to show that the contents of the file on disk matches what's in the editor.

It's a little round-about, but it avoids the indicator getting out of sync if we and another process write the file simultaneously.

Is it possible that inotify does not work on the drive you're saving to?

**Comment by notpeter** [comment]
Following up on this. What filesystem are you running on the external drive and with what mount options?

I think it is likely this mount does not support inotify -- if that is the case, Zed cannot currently track changes without periodically rescanning your entire project.
